### PR TITLE
[libc][docgen] support non-top-level headers

### PR DESCRIPTION
**Pull request by nickdesaulniers**
such as arpa/inet, sys/*

### CODE REVIEW

File: libc/utils/docgen/docgen.py [hunk 1]
```diff
@@ -176,10 +180,22 @@ def print_impl_status_rst(header: Header, api: Dict):
         print_functions_rst(header, api["functions"])
 
 
+# This code implicitly relies on docgen.py being in the same dir as the json
+# files and is likely to need to be fixed when re-integrating docgen into
+# hdrgen.
+def get_choices():
```
Comment (lntue) [inline libc/utils/docgen/docgen.py:186]: nit:
```
def get_choices() -> List:
```

Comment (nickdesaulniers) [reply libc/utils/docgen/docgen.py]: ba1d7489eb30

good catch, I'm still not used to putting explicit types in my python...

**Review by lntue — APPROVED** [review]

File: libc/utils/docgen/docgen.py [hunk 2]
```diff
@@ -176,10 +180,22 @@ def print_impl_status_rst(header: Header, api: Dict):
         print_functions_rst(header, api["functions"])
 
 
+# This code implicitly relies on docgen.py being in the same dir as the json
+# files and is likely to need to be fixed when re-integrating docgen into
+# hdrgen.
+def get_choices() -> List:
```
Comment (nickdesaulniers) [inline libc/utils/docgen/docgen.py:186]:
```
Traceback (most recent call last):
  File "/android0/llvm-project/./libc/utils/docgen/docgen.py", line 186, in <module>
    def get_choices() -> List:
                         ^^^^
NameError: name 'List' is not defined. Did you mean: 'list'?
```
sigh...

Comment (nickdesaulniers) [reply libc/utils/docgen/docgen.py:186]: 52db903888eace2e4053a751c8f058ac7c98b49d

**Pull request merged**